### PR TITLE
nixos/nexus: add jvm package option

### PR DESCRIPTION
**Pull request by TomatoCream**
###### Motivation for this change
Nexus does not run without jvm directory

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/web-apps/nexus.nix [hunk 1]
```diff
@@ -19,6 +17,16 @@ in
         description = "Package which runs Nexus3";
       };
 
+      jdkPackage = mkOption {
+        type = types.package;
+        default = pkgs.openjdk8;
+        defaultText = literalExample "pkgs.openjdk8";
+        example = literalExample "pkgs.openjdk8";
+        description = ''
+          The jdk package to use
```
Comment (fufexan) [inline nixos/modules/services/web-apps/nexus.nix:26]:
```suggestion
          The JDK package to use.
```

File: nixos/modules/services/web-apps/nexus.nix [hunk 2]
```diff
@@ -6,9 +6,7 @@ let
 
   cfg = config.services.nexus;
 
-in
```
Comment (fufexan) [inline nixos/modules/services/web-apps/nexus.nix:9]: Please make sure to format with `nixpkgs-fmt`.

**Review by fufexan — CHANGES_REQUESTED** [review]

File: nixos/modules/services/web-apps/nexus.nix (hunk 2 above)
Comment (TomatoCream) [reply nixos/modules/services/web-apps/nexus.nix:9]: oh I did not know it was different from nixfmt

**Comment by roberth** [comment]
Hi @TomatoCream, thank you for making this PR.

I'm surprised to see that [`nixosTests.nexus`](https://github.com/NixOS/nixpkgs/blob/8284fc30c84ea47e63209d1a892aca1dfcd6bdf3/nixos/tests/nexus.nix) doesn't catch this problem. It is quite a simple test. How can it be improved to detect configuration problems like this in the future?

Could you please configure your `git` with your (user)name and an email address that you've added to your GitHub personal email settings? Otherwise, the GitHub UI won't show your commits properly.

**Review by fufexan — APPROVED** [review]

**Review by ironpinguin — APPROVED** [review]

**Review by Janik-Haag — APPROVED** [review]
This looks good to me, @TomatoCream are you still interested in getting this merged and is this still relevant. Sorry for no one merging this earlier :( feel free to ping me in the future.

**Comment by Janik-Haag** [comment]
One small thing if you want this to be merged you have to change the commit message to something like `nexus: add jvm package` so it follows `contributing.md`

File: nixos/modules/services/web-apps/nexus.nix (hunk 2 above)
Comment (Janik-Haag) [reply nixos/modules/services/web-apps/nexus.nix:9]: There is currently no official formater for nixpkgs despite the name of nixpkgs-fmt. There is a formatting rfc in progress and the reference implementation can be found here:  https://github.com/serokell/nixfmt/pull/118

**Comment by Janik-Haag** [comment]
I took the liberty of renaming the commit since the author didn't respond.

File: nixos/modules/services/web-apps/nexus.nix [hunk 3]
```diff
@@ -19,6 +18,16 @@ in
         description = "Package which runs Nexus3";
       };
 
+      jdkPackage = mkOption {
```
Comment (SuperSandro2000) [inline nixos/modules/services/web-apps/nexus.nix:21]: @Janik-Haag when merging old PRs, please make sure they are up to standard. This option triggers the following warning ``trace: warning: literalExample is deprecated, use literalExpression instead, or use literalMD for a non-Nix description.`` and should use mkPackageOption.

Comment (SuperSandro2000) [reply nixos/modules/services/web-apps/nexus.nix:21]: #244323

Comment (Janik-Haag) [reply nixos/modules/services/web-apps/nexus.nix:21]: Sorry, I missed that, thank you for pointing this out.

**Pull request merged**